Sources/ReleaseServer/ReleaseServer.WebApi/Controllers/ReleaseArtifactController.cs

```diff
@@ -131,31 +196,65 @@ public async Task<IActionResult>  GetLatestArtifact([Required] string product, [
             return new FileContentResult(response.Payload, contentType);
         }
 
+        /// <summary>
+        /// Retrieves the latest version of a specific product.
+        /// </summary>
+        /// <param name="product"></param>
+        /// <param name="os"></param>
+        /// <param name="architecture"></param>
+        /// <response code="200">The specified product exists.</response>
+        /// <response code="500">The product is not available for the specified platform (OS + HW architecture)
+        /// or the product with the specified product name does not exist</response>
         [AllowAnonymous]
         [HttpGet("latest/{product}/{os}/{architecture}")]
+        [ProducesResponseType(typeof(ProductVersionResponseModel), 200)]
         public async Task<ProductVersionResponseModel> GetLatestVersion([Required] string product, [Required] string os, [Required] string architecture)
         {
             var latestVersion = await ReleaseArtifactService.GetLatestVersion(product, os, architecture);
 
             return latestVersion.ToProductVersionResponse();
         }
         
+        /// <summary>
+        /// Deletes the specified product.
+        /// </summary>
+        /// <param name="product"></param>
+        /// <param name="os"></param>
+        /// <param name="architecture"></param>
+        /// <param name="version"></param>
+        /// <response code="200">The specified product got deleted successfully.</response>
+        /// <response code="401">The user is not authorized (wrong credentials or missing auth header).</response>
+        /// <response code="500">There exists no product with the specified product name.</response>
         [HttpDelete("{product}/{os}/{architecture}/{version}")]
-        public IActionResult DeleteSpecificArtifact ([Required] string product, [Required] string os, [Required] string architecture, [Required] string version)
+        public async Task<IActionResult> DeleteSpecificArtifact ([Required] string product, [Required] string os, [Required] string architecture, [Required] string version)
         {
-            ReleaseArtifactService.DeleteSpecificArtifact(product, os, architecture, version);
+            await ReleaseArtifactService.DeleteSpecificArtifact(product, os, architecture, version);
 
             return Ok("artifact successfully deleted");
         }
         
+        /// <summary>
+        /// Deletes all products of a specific product name.
+        /// </summary>
+        /// <param name="product"></param>
+        /// <returns></returns>
+        /// <response code="200">All products of the specified product name got deleted successfully.</response>
+        /// <response code="401">The user is not authorized (wrong credentials or missing auth header).</response>
+        /// <response code="500">There exists no product with the specified product name.</response>
```

Comment:
Same as above: A 500 error code represents a server-side error. 404 (Not Found) would be more appropriate here.

Reply:
Fixed with the same showed above

Reply:
see https://github.com/Traeger-GmbH/release-server/pull/50/commits/89c4e62a1637ac6d990d290b01c6e126d5ef6900